### ISSUE TITLE
代码错误？

### ISSUE THREAD

**Issue by AndrewTal** [WeijiaZhang24/CausalMIL#3]
class auxiliary_y_fixed(nn.Module):
***
        for iter_id, bag in enumerate(bags):
            bag_id = bag.item()
            instances_bag = bag_idx.eq(bag_id).nonzero().squeeze()
            if instances_bag.numel()>0:
                if instances_bag.numel()>1:
                    M_bag, index = torch.max(loc_ins[instances_bag],dim=0)
                    M[iter_id, :] = M_bag
                    max_z_ins[iter_id, :] = z_ins[index]
                    max_instances[iter_id, :] = bag_instances[index]
                    max_instances_mu[iter_id, :] = instance_mu[index]
                    max_instances_std[iter_id, :] = instance_std[index]
***
                    
loc_ins[instances_bag] 可以获取 iter_id bag内所有的 instance，torch.max后可以获得bag内概率最大的样本的 index，这个 index 是bag 内的 index，而非全部 instance 的 index。

但是后面赋值时，z_ins[index]、bag_instances[index]、instance_mu[index]、instance_std[index] 这里需要的应该是全局（全部instance）的index？是不是有明显的错误，还是我理解有错误？

**Comment by WeijiaZhang24** [comment]
For each bag, "index" refers to "bag内概率最大的样本的 index". If it is a batch of bags, "index" still refers to "bag内概率最大的样本的 index" for each bag.

**Comment by AndrewTal** [comment]
> For each bag, "index" refers to "bag内概率最大的样本的 index". If it is a batch of bags, "index" still refers to "bag内概率最大的样本的 index" for each bag.


If it is a batch of bags, the code should be:
 
M_bag, index = torch.max(loc_ins[instances_bag],dim=0)
M[iter_id, :] = M_bag
max_z_ins[iter_id, :] = z_ins[instances_bag][index]
max_instances[iter_id, :] = bag_instances[instances_bag][index]
max_instances_mu[iter_id, :] = instance_mu[instances_bag][index]
max_instances_std[iter_id, :] = instance_std[instances_bag][index]

**Comment by WeijiaZhang24** [comment]
I see your point. This is indeed an unintended mistake. The original code only selects the correct "index" for the first bag.
